sklearn/decomposition/_nmf.py

```diff
@@ -1179,7 +1194,19 @@ def __init__(
     def _check_params(self, X):
         # n_components
         self._n_components = self.n_components
-        if self._n_components is None:
+        if self.n_components == "warn":
+            warnings.warn(
+                (
+                    "The default value of `n_components` will change from `None` to"
+                    " `'auto'` in 1.6. Set the value of `n_components` to `None`"
+                    " explicitly to supress the warning"
+                ),
+                FutureWarning,
+            )
+            self._n_components = None  # Keeping the old default value
+        if self._n_components is None or (
+            self.init != "custom" and self.n_components == "auto"
```

Comment:
```suggestion
            self.init != "custom" and self._n_components == "auto"
```

Reply:
fixed in b3529bb1a0b276990b2124907d9b2f9b8f790b5d

Reply:
fixed in b3529bb1a0b276990b2124907d9b2f9b8f790b5d